### DISCUSSION TITLE
Is there anyway of getting refs to work without attaching them to the root store?

### DISCUSSION THREAD

**Discussion by maccman** [xaviergonz/mobx-keystone#481] [Q&A] [answered]
Some context we use keystone in two ways:

- models, holding our data structure in memory
- model views, basically fancy views that get logic out of the React components and into a nice class like interface.

We'd rather not attach these model views to the root-store because we'd prefer to instantiate them in the react components (to decouple them from the models).

so is there anyway of getting refs to work without attaching them to the root store?

**Comment by xaviergonz** [comment, marked as answer]
You could use custom references that actually use the other root store to resolve the ids

https://mobx-keystone.js.org/references#custom-references
